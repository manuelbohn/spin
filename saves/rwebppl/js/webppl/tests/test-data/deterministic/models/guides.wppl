var numParamsCreatedBy = function(thunk) {
  setFreshParamsId();
  thunk();
  return _.size(getParams());
};

[
  param({mu: 1, sigma: 0}) === 1,
  param({init: ones}) === 1,
  T.sumreduce(param({dims: [3, 2], mu: 1, sigma: 0})) === 6,

  // Check (indirectly) that a guide is automatically generated, by
  // checking that a parameter is created.

  numParamsCreatedBy(function() {
    Infer({method: 'SMC', particles: 1, importance: 'default', model() {
      return flip();
    }});
  }) === 0,

  numParamsCreatedBy(function() {
    Infer({method: 'SMC', particles: 1, importance: 'autoGuide', model() {
      return flip();
    }});
  }) === 1,

  numParamsCreatedBy(function() {
    Infer({method: 'SMC', particles: 1, importance: 'autoGuide', model() {
      return sample(Bernoulli({p: 0.5}), {noAutoGuide: true});
    }});
  }) === 0,

  numParamsCreatedBy(function() {
    Infer({method: 'forward', samples: 1, guide: true, model() {
      return sample(Bernoulli({p: 0.5}));
    }});
  }) === 1,

  numParamsCreatedBy(function() {
    Infer({method: 'forward', samples: 1, guide: true, model() {
      return sample(Bernoulli({p: 0.5}), {noAutoGuide: true});
    }});
  }) === 0
];
