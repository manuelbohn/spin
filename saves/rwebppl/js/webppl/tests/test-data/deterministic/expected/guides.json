{
  "result": [true, true, true, true, true, true, true, true]
}
